### PR TITLE
Update esi sdk

### PR DESCRIPTION
**Pull request by DanNiESh**
Related issue: https://github.com/CCI-MOC/esi/issues/536

### CODE REVIEW

**Comment by tzumainn** [comment]
Looks good! Can you update the description of the PR to be a bit more specific about the contents - e.g., "updates to object attributes, addition of functional tests"?

**Comment by DanNiESh** [comment]
> Looks good! Can you update the description of the PR to be a bit more specific about the contents - e.g., "updates to object attributes, addition of functional tests"?

Done! I'm now working on python-esileapclient PR. I might update this PR slightly (attribute naming, etc.) to sync with python-esileapclient if necessary. I 'll let you know when this one is in final stage to review.

**Comment by tzumainn** [comment]
Okay, sounds good!

**Review by tzumainn — CHANGES_REQUESTED** [review]
Looks good! I think it could use one thing: in the README, could you add a quick explanation of how you'd use this sdk within a python script?

**Comment by DanNiESh** [comment]
I've updated the README.md, let me know what you think!

File: README.md [hunk 1]
```diff
@@ -1,2 +1,68 @@
...
+
+### Create a connection to ESI SDK
+
+There are several methods to establish a connection using the ESI SDK. Since the ***esi.connection.ESIConnection*** class inherits from ***openstack.connection.Connection***, [methods](https://docs.openstack.org/openstacksdk/latest/user/connection.html) applicable for creating connections in the OpenStack SDK can also be used with the ESI SDK. Below are some common ways to create an ESIConnection:
+
+**Create a connection using only keyword arguments**
+```
+from esi import connection
+
+conn = connection.Connection(
```
Comment (tzumainn) [inline README.md]: I think this is actually ESIConnection

File: README.md [hunk 2]
```diff
@@ -1,2 +1,68 @@
...
+**Create a connection from existing CloudRegion**
+```
+from esi import connection
+import openstack.config
+
+config = openstack.config.get_cloud_region(
+    cloud='example',
+    region_name='earth',
+)
+conn = connection.Connection(config=config)
```
Comment (tzumainn) [inline README.md]: Here as well

**Review by tzumainn — CHANGES_REQUESTED** [review]
Thanks for adding the README! The code works well; there's just a few minor inaccuracies in the documentation that I've noted.

Could you also add a `connect` method, similar to https://github.com/openstack/openstacksdk/blob/master/openstack/__init__.py? What this allows one to do is have code like:

```
import openstack

conn = openstack.connect()
nodes = conn.baremetal.nodes(details=True)
```

And the openstacksdk will simply pull from your current environment.

**Comment by DanNiESh** [comment]
> Thanks for adding the README! The code works well; there's just a few minor inaccuracies in the documentation that I've noted.
> 
> Could you also add a `connect` method, similar to https://github.com/openstack/openstacksdk/blob/master/openstack/__init__.py? What this allows one to do is have code like:
> 
> ```
> import openstack
> 
> conn = openstack.connect()
> nodes = conn.baremetal.nodes(details=True)
> ```
> 
> And the openstacksdk will simply pull from your current environment.

Done!

**Comment by tzumainn** [comment]
This looks great - thanks!

**Review by tzumainn — APPROVED** [review]

**Pull request merged**